components/esp32/deep_sleep.c

```diff
@@ -318,18 +318,24 @@ uint64_t esp_deep_sleep_get_ext1_wakeup_status()
     }
     uint32_t status = REG_GET_FIELD(RTC_CNTL_EXT_WAKEUP1_STATUS_REG, RTC_CNTL_EXT_WAKEUP1_STATUS);
     // Translate bit map of RTC IO numbers into the bit map of GPIO numbers
-    uint64_t gpio_mask = 0;
-    for (int gpio = 0; gpio < GPIO_PIN_COUNT; ++gpio) {
+    uint64_t gpio_mask = 1;
+    uint64_t gpio_return = 0;
+    for (unsigned int gpio = 0; gpio < GPIO_PIN_COUNT; ++gpio) {
+
+        if(gpio > 0) {
+            gpio_mask <<= 0x1;
+        }
+
         if (!RTC_GPIO_IS_VALID_GPIO(gpio)) {
             continue;
         }
         int rtc_pin = rtc_gpio_desc[gpio].rtc_num;
         if ((status & BIT(rtc_pin)) == 0) {
             continue;
         }
-        gpio_mask |= BIT(gpio);
```

Comment:
Why? If gpio = 0, BIT(0) = 1, and gpio_mask = 1.  Are you sure you aren't misunderstanding the macro BIT?  The patch does not seem like it would work.  Igrr's suggestion seems right to fix gpio >= 32.